### PR TITLE
Fix Floating Grouped Windows Crash and Bug.

### PR DESCRIPTION
**Pull request by The-Briel-Deal**
#### Describe your PR, what does it fix/add?
It was reported here https://github.com/hyprwm/Hyprland/issues/5768 that floating groups would display as two separate windows then Hyprland would eventually hang after moving or resizing. After further testing this only happens with minsize, or maxsize set in config. This was due to the max/min size check forcing windows to unhide.

Since I'm not sure why this was done in the first place I only added a check verifying that its not unhiding grouped windows here (nextWindow will be a null pointer only if the window is not grouped).

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
I'm unsure why we force unhid windows in a size check so there could be a potential side effect. That said, I did test this and this only effects grouped windows.

#### Is it ready for merging, or does it need work?
I believe so, that said there may be a better way to do this.

### CODE REVIEW

**Review by vaxerski — COMMENTED** [review]
Yeah I think it was a remnant from `size`. It should only be in `minsize` (in maxsize it makes no sense)

The thing is we'd set 0x0 windows that xwayland sometimes creates for no reason to hidden to avoid a small border being drawn around a microscopic speck.

File: src/desktop/Window.cpp [hunk 1]
```diff
@@ -707,7 +707,8 @@ void CWindow::applyDynamicRule(const SWindowRule& r) {
             m_vRealSize                     = Vector2D(std::min((double)m_sAdditionalConfigData.maxSize.toUnderlying().x, m_vRealSize.goal().x),
                                                        std::min((double)m_sAdditionalConfigData.maxSize.toUnderlying().y, m_vRealSize.goal().y));
             g_pXWaylandManager->setWindowSize(m_pSelf.lock(), m_vRealSize.goal());
-            setHidden(false);
+            if (!m_sGroupData.pNextWindow.lock())
```
[the remark this answers is not in the document]
Comment (The-Briel-Deal) [reply src/desktop/Window.cpp:710]: Oh nice! I had no idea what expired did until I read this! Thanks for the pointer :^) !

Fixing now!

Also, from my reading of this, it sounds like you are saying I can drop the second setHidden? I wonder if there was a reason for that, looking at git blame now.

[the remark this answers is not in the document]
Comment (vaxerski) [reply src/desktop/Window.cpp]: yea you can drop it from `maxsize`. Leave it in the maxsize

**Comment by vaxerski** [comment]
you can drop the sethidden entirely from maxsize. it's unintended imo

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**